### DISCUSSION TITLE
May I write a latex renderer for goldmark?

### DISCUSSION THREAD

**Discussion by soypat** [yuin/goldmark#322] [Ideas]
Hello! I want to write a $\LaTeX$ renderer for goldmark! I've been using blackfriday but its pretty broken. 

The renderer I've written for blackfriday is over here: https://github.com/soypat/exercises/blob/main/md2latex/latex.go. Was thinking of porting it over to goldmark and was wondering if
* It could be added as a renderer under a directory in `renderer/latex` directory
* Maybe some directions on how to best proceed- should I implement `renderer.Renderer` interface? I noticed the html renderer implements `NodeRenderer` interface over this other one.

Thanks for this awesome project!

**Comment by yuin** [comment]
goldmark is an extensible library. You can write your own renderers an external library instead of `renderer/latex` like this

https://github.com/stephenafamo/goldmark-pdf

**Reply by soypat** [reply]
Ok, fair enough! Would you suggest I base myself on the html renderer implementation for building the latex renderer?

**Comment by soypat** [comment]
@yuin Got it working! https://github.com/soypat/goldmark-latex

Any chance it could be added to the list of extensions available?
